### PR TITLE
Better docs

### PR DESCRIPTION
**Pull request by bjorn3**
Working on more doc improvements

Edit: done for today

### CODE REVIEW

File: src/librustc/infer/region_inference/graphviz.rs [hunk 1]
```diff
@@ -133,7 +133,6 @@ enum Node {
     Region(ty::RegionKind),
 }
 
-// type Edge = Constraint;
```
Comment (bjorn3) [inline src/librustc/infer/region_inference/graphviz.rs]: Leftover?

**Comment by bjorn3** [comment]
Ready for review

**Comment by frewsxcv** [comment]
```
tidy check (x86_64-unknown-linux-gnu)
[00:02:18] tidy error: /checkout/src/librustc/hir/def.rs:166: trailing whitespace
[00:02:18] tidy error: /checkout/src/librustc/hir/lowering.rs:73: trailing whitespace
[00:02:18] tidy error: /checkout/src/librustc/hir/lowering.rs:76: trailing whitespace
[00:02:18] tidy error: /checkout/src/librustc/infer/region_inference/mod.rs:232: trailing whitespace
[00:02:20] some tidy checks failed
```

**Comment by bjorn3** [comment]
Fixed tidy errors

**Comment by frewsxcv** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**